graphql-cli/src/main/java/io/ballerina/graphql/generator/CodeGenerator.java

```diff
@@ -119,16 +116,10 @@ private List<SrcFilePojo> generateBalSources(GraphqlProject project) throws Clie
     private void generateClients(String projectName, List<String> documents, GraphQLSchema schema,
                                  AuthConfig authConfig, List<SrcFilePojo> sourceFiles)
             throws ClientGenerationException, IOException {
-        for (String document : documents) {
-            File documentFile = new File(document);
-            Document queryDocument = Utils.getGraphQLQueryDocument(document);
-            String queryDocumentName = CodeGeneratorUtils.getDocumentName(documentFile);
-
-            String clientSrc = ClientGenerator.getInstance().
-                    generateSrc(queryDocument, queryDocumentName, schema, authConfig);
-            sourceFiles.add(new SrcFilePojo(SrcFilePojo.GenFileType.GEN_SRC, projectName,
-                    CodeGeneratorUtils.getClientFileName(documentFile), clientSrc));
-        }
+        String clientSrc = ClientGenerator.getInstance().
+                generateSrc(documents, schema, authConfig);
+        sourceFiles.add(new SrcFilePojo(SrcFilePojo.GenFileType.GEN_SRC, projectName,
+                "client.bal", clientSrc));
```

Comment:
Let's define the client file name (i.e. client.bal) as a constant in the `CodeGeneratorConstants.java` file.